core/block/editor/table/editor.go

```diff
@@ -0,0 +1,781 @@
+	})
+	if !s.Add(rowsLayout) {
+		return "", fmt.Errorf("add rows block")
+	}
+
+	tableBlock.Model().ChildrenIds = []string{columnsLayout.Model().Id, rowsLayout.Model().Id}
+
+	if req.WithHeaderRow {
+		if len(rowIDs) == 0 {
+			return "", fmt.Errorf("no rows to make header row")
+		}
+		headerID := rowIDs[0]
+
+		if err := t.RowSetHeader(s, pb.RpcBlockTableRowSetHeaderRequest{
+			TargetId: headerID,
+			IsHeader: true,
+		}); err != nil {
+			return "", fmt.Errorf("row set header: %w", err)
+		}
+
+		if err := t.RowListFill(s, pb.RpcBlockTableRowListFillRequest{
+			BlockIds: []string{headerID},
+		}); err != nil {
+			return "", fmt.Errorf("fill header row: %w", err)
+		}
+
+		row, err := getRow(s, headerID)
+		if err != nil {
+			return "", fmt.Errorf("get header row: %w", err)
+		}
+
+		for _, cellID := range row.Model().ChildrenIds {
+			cell := s.Get(cellID)
+			if cell == nil {
+				return "", fmt.Errorf("get header cell id %s", cellID)
+			}
+
+			cell.Model().BackgroundColor = "grey"
+		}
+	}
+
+	return tableBlock.Model().Id, nil
+}
+
+func (t *editor) CellCreate(s *state.State, rowID string, colID string, b *model.Block) (string, error) {
+	tb, err := NewTable(s, rowID)
+	if err != nil {
+		return "", fmt.Errorf("initialize table state: %w", err)
+	}
+
+	row, err := getRow(s, rowID)
+	if err != nil {
+		return "", fmt.Errorf("get row: %w", err)
+	}
+	if _, err = pickColumn(s, colID); err != nil {
+		return "", fmt.Errorf("pick column: %w", err)
+	}
+
+	cellID, err := addCell(s, rowID, colID)
+	if err != nil {
+		return "", fmt.Errorf("add cell: %w", err)
+	}
+	cell := s.Get(cellID)
+	cell.Model().Content = b.Content
+	if err := s.InsertTo(rowID, model.Block_Inner, cellID); err != nil {
+		return "", fmt.Errorf("insert to: %w", err)
+	}
+
+	colIdx := tb.MakeColumnIndex()
+	normalizeRow(nil, colIdx, row)
+
+	return cellID, nil
+}
+
+func (t *editor) RowCreate(s *state.State, req pb.RpcBlockTableRowCreateRequest) (string, error) {
+	switch req.Position {
+	case model.Block_Top, model.Block_Bottom:
+	case model.Block_Inner:
+		tb, err := NewTable(s, req.TargetId)
+		if err != nil {
+			return "", fmt.Errorf("initialize table state: %w", err)
+		}
+		req.TargetId = tb.Rows().Id
+	default:
+		return "", fmt.Errorf("position is not supported")
+	}
+
+	rowID, err := t.addRow(s)
+	if err != nil {
+		return "", err
+	}
+	if err := s.InsertTo(req.TargetId, req.Position, rowID); err != nil {
+		return "", fmt.Errorf("insert row: %w", err)
+	}
+	return rowID, nil
+}
+
+func (t *editor) RowDelete(s *state.State, req pb.RpcBlockTableRowDeleteRequest) error {
+	_, err := pickRow(s, req.TargetId)
+	if err != nil {
+		return fmt.Errorf("pick target row: %w", err)
+	}
+
+	if !s.Unlink(req.TargetId) {
+		return fmt.Errorf("unlink row block")
+	}
+	return nil
+}
+
+func (t *editor) RowDuplicate(s *state.State, req pb.RpcBlockTableRowDuplicateRequest) (newRowID string, err error) {
+	if req.Position != model.Block_Top && req.Position != model.Block_Bottom {
```

Comment:
I have added this check to exclude insertion of new Row block in some wrong place relatively to old row